### ISSUE TITLE
Nested ranges: Vertical table is not generated inside horizontal

### ISSUE THREAD

**Issue by kamil-szprych** [ClosedXML/ClosedXML.Report#136]
There is example of horizontal table here: https://closedxml.github.io/ClosedXML.Report/docs/en/Flat-tables.html but its the other way around in my case.

In example above data was arranged in a way that every entry on the list was a row and each row had array and that array was this dynamic length horizontal element. The problem is that my data structure is a little bit different, essentialy its list of columns. Its list of objects that have header and list of objects that are column data. Im not sure how to make template so it will iterate horizontally through entities and then for each of them vertically through subentities. The template i made just generates horizontal data but nothing else and there is no errors.

Thats the main problem. Additionaly there is still another issue. Sub entities consists of name and quantity. Every parent entity has the same list of names in subentity, just diffrent quantities.

To visualize it, lets say i have list of orders where every order is separate column and they consists of the same items just with diffrent quantities.

Order structure:

```
class Order
  {
     public string OrderNumber{ get; set; }
     public IEnumerable<ProductWithQuantity> ProductsWithQuantities{ get; set; }
  }

class ProductWithQuantity
  {
     public string ProductName{ get; set; }
     public int Quantity{ get; set; }
  }
```

I want to pass `IEnumerable<Order>` and using template generate excel like this:
![image](https://user-images.githubusercontent.com/51128962/79863073-6db4e380-83d7-11ea-9323-9280f065aa0f.png)

I created range with `<<Range Horizontal>>` tag and inside it i created another range entity_subentity where i have `{{item.Quantity}}` but it does not work, only horizontal OrderNumbers are generated and they have many empty cells between them.

**Comment by kamil-szprych** [comment]
Simple test template (for now with ProductName repeated for every column):
[SimpleTemplate.xlsx](https://github.com/ClosedXML/ClosedXML.Report/files/4509843/SimpleTemplate.xlsx)

**Comment by b0bi79** [comment]
@kamil-szprych 
1. You have created a template that does not meet the requirements. There is not enough service column. [See documentation](https://closedxml.github.io/ClosedXML.Report/docs/en/Flat-tables.html#vertical-tables)
>Each range specifying a vertical table must have at least two columns and two rows. The leftmost column and bottommost row serve to hold configuration information and are treated specially. After the report is built the service column is cleared, and the service row is deleted if it is empty.

2. I think that using a pivot table is more suitable for solving this problem. [Documentation](https://closedxml.github.io/ClosedXML.Report/docs/en/Pivot-tables)

**Comment by b0bi79** [comment]
Done horizontal inside vertical
![image](https://user-images.githubusercontent.com/1150085/126066293-c92244a4-ef9f-4edf-a2ab-1d9dfedc63ea.png)

**Comment by b0bi79** [comment]
Closed with #221.
Fixed the function of generating horizontal tables inside vertical ones. It is still impossible to generate vertical tables inside horizontal ones.

**Comment by Vazovsk1y** [comment]
> Closed with #221. Fixed the function of generating horizontal tables inside vertical ones. It is still impossible to generate vertical tables inside horizontal ones.

@b0bi79 
At the moment there is still no way to generate vertical tables inside horizontal?
